open-craft/section-to-course
# feat: Initial copy-paste circuit

This PR creates the initial scaffolding for the section_to_course application.

**JIRA tickets**: https://tasks.opencraft.com/browse/BB-7295

**Dependencies**: https://github.com/openedx/edx-platform/pull/32058

**Testing instructions**:

To set up the app:

1. Check out the edx-platform branch as linked in the Dependencies section
2. In the `src` directory of the devstack, clone this repository and checkout this branch.
3. In the `devstack` repo root, run `make dev.shell.studio`
4. `cd /edx/src/section_to_course`
5. `pip install -e .`

To run the unit tests:

7. From the `devstack` repo root, run `make dev.shell.studio`
8. `DJANGO_SETTINGS_MODULE=cms.envs.test pytest --pyargs section_to_course --rootdir cms`

To test the functionality manually:

1. Create a course (or find an existing one) and pick a section from it. Derive the usage key string (this can be done, for instance, by duplicating a block and checking the network request in the devtools to see what the source block string is)
2. From the `devstack` repo root, run `make dev.shell.studio`
3. Run the management command to perform the copy. The command is `section_to_course` and should be followed by the destination course and the source section's usage key. For example, `./manage.py cms section_to_course course-v1:Beeper+B33P+BOOP block-v1:edX+DemoX+Demo_Course+type@chapter+block@graded_interactions edx`.

**Author notes and concerns**:

The cookiecutter we derived from had a lot of nice CI functionalities, not to mention linting. I had to discard all of it since we rely on the edx-platform environment and can't (readily) import that into the GitHub CI. So, please be attentive to code quality when reviewing.
